packages/core/src/indexing-store/readonly.ts

```diff
@@ -116,7 +120,9 @@ export const getReadonlyStore = ({
       if (after === null && before === null) {
         query = query.limit(limit + 1);
         const rows = await query.execute();
-        const records = rows.map((row) => decodeRow(row, table, kind));
+        const records = rows.map((row) =>
+          decodeRecord({ record: row, table, encoding }),
+        );
```

Comment:
I wonder if there's an opportunity to always use "row" for database-compatible objects (either just before inserting or just after selecting) and "record" for user-land objects. If yes, I think it would make sense for this to look like `decodeRow({ row, table, encoding })`.